resources/installation.rb

```diff
@@ -37,7 +37,7 @@ def binary_basename
   archive_path = node.join_path(Chef::Config[:file_cache_path], basename)
   remote_file archive_path do
     source format(attrs['archive_url'], version: new_resource.version, basename: basename)
-    checksum node.binary_checksum(node, new_resource.version)
+    checksum attrs['archive_checksum'] || nil
```

Comment:
do you think the `|| nil` is necessary here? I'd have no issue with making the checksum mandatory, it was the case before

Reply:
the remote_file function handles nil checksums. I'd say we could enforce a checksum on our side as FAR. I'm currently trying to mimic the old behavior with a chef-remote-file deployment though.

```
    def self.remote_file(node, cookbook, file, checksum = nil)
      if checksum.nil?
        ::Chef::Log.warn "The file #{file} has no checksum. This is a poor practice."
      else
        file = "#{checksum}-#{file}"
      end
      ::File.join(node['criteo_infra']['filer']['url'], cookbook, file)
    end
```